salt/cloud/clouds/gogrid.py

```diff
@@ -281,3 +163,367 @@ def create(vm_):
+
+        salt-cloud -f list_public_ips <provider>
+
+    To list unavailable (assigned) IPs, use:
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt-cloud -f list_public_ips <provider> state=assigned
+    '''
+    if kwargs is None:
+        kwargs = {}
+
+    args = {}
+    if 'state' in kwargs:
+        if kwargs['state'] == 'assigned':
+            args['ip.state'] = 'Assigned'
+        else:
+            args['ip.state'] = 'Unassigned'
+    else:
+        args['ip.state'] = 'Unassigned'
+
+    args['ip.type'] = 'Public'
+
+    response = _query('grid', 'ip/list', args=args)
+
+    ret = {}
+    for item in response['list']:
+        name = item['ip']
+        ret[name] = item
+
+    return ret
+
+
+def list_common_lookups(kwargs=None, call=None):
+    '''
+    List common lookups for a particular type of item
+    '''
+    if kwargs is None:
+        kwargs = {}
+
+    args = {}
+    if 'lookup' in kwargs:
+        args['lookup'] = kwargs['lookup']
+
+    response = _query('common', 'lookup/list', args=args)
+
+    return response
+
+
+def destroy(name, call=None):
+    '''
+    Destroy a machine by name
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -d vm_name
+
+    '''
+    if call == 'function':
+        raise SaltCloudSystemExit(
+            'The destroy action must be called with -d, --destroy, '
+            '-a or --action.'
+        )
+
+    salt.utils.cloud.fire_event(
+        'event',
+        'destroying instance',
+        'salt/cloud/{0}/destroying'.format(name),
+        {'name': name},
+        transport=__opts__['transport']
+    )
+
+    response = _query('grid', 'server/delete', args={'name': name})
+
+    salt.utils.cloud.fire_event(
+        'event',
+        'destroyed instance',
+        'salt/cloud/{0}/destroyed'.format(name),
+        {'name': name},
+        transport=__opts__['transport']
+    )
+
+    if __opts__.get('update_cachedir', False) is True:
+        salt.utils.cloud.delete_minion_cachedir(name, __active_provider_name__.split(':')[0], __opts__)
+
+    return response
+
+
+def reboot(name, call=None):
+    '''
+    Reboot a machine by name
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -a reboot vm_name
+    '''
+    return _query('grid', 'server/power', args={'name': name, 'power': 'restart'})
+
+
+def stop(name, call=None):
+    '''
+    Stop a machine by name
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -a stop vm_name
+    '''
+    return _query('grid', 'server/power', args={'name': name, 'power': 'stop'})
+
+
+def start(name, call=None):
+    '''
+    Start a machine by name
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -a start vm_name
+    '''
+    return _query('grid', 'server/power', args={'name': name, 'power': 'start'})
+
+
+def show_instance(name, call=None):
+    '''
+    Start a machine by name
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -a start vm_name
```

Comment:
Nothing. I was commenting on an old page and my comment accidentally ended up here, instead of on the other one.